addons/purchase/models/account_invoice.py

```diff
@@ -79,6 +79,11 @@ def _onchange_purchase_auto_complete(self):
     @api.onchange('partner_id', 'company_id')
     def _onchange_partner_id(self):
         res = super(AccountMove, self)._onchange_partner_id()
+        delivery_partner_id = self._get_invoice_delivery_partner_id()
+        new_fiscal_position_id = self.env['account.fiscal.position'].with_context(force_company=self.company_id.id).get_fiscal_position(self.partner_id.id, delivery_id=delivery_partner_id)
+        self.fiscal_position_id = self.purchase_id.fiscal_position_id \
+            or self.env['purchase.order'].browse(self.env.context.get('default_purchase_id')).fiscal_position_id \
+            or self.env['account.fiscal.position'].browse(new_fiscal_position_id)
```

Comment:
I don't understand why this would be in a onchange. If you create a `account.move` by setting both a `partner_id` and a `fiscal_position_id`, the later should not be overriden.
It needs to be overriden if you change the `partner_id` afterwards though.

Reply:
Because of the 'company_id', the _onchange_partner_id (https://github.com/odoo/odoo/blob/27edc28a2c19fad80053e668a8c72e7a629e4c1a/addons/purchase/models/account_invoice.py#L80) is called again after _onchange_purchase_auto_complete (https://github.com/odoo/odoo/blob/27edc28a2c19fad80053e668a8c72e7a629e4c1a/addons/purchase/models/account_invoice.py#L27) and, the fiscal  position is overriden.

Reply:
Ideally the onchange would depend on partner_shipping_id defined in sales.  (we moved it to account in master, so that would be good to check in the fw-port)

Reply:
By the way, in purchase_stock  and in https://github.com/odoo/odoo/blob/d9ff2e8ec7200aa15830e92e5083251c4e81dd1c/addons/purchase/models/purchase.py#L240 it should take that into account as well then.  (I remember a discussion where people wanted to remove the field simply from the view)

Reply:
That's in the purchase_order, but when you create bill from the PO, you pass into the _onchange_partner_id here https://github.com/odoo/odoo/blob/27edc28a2c19fad80053e668a8c72e7a629e4c1a/addons/account/models/account_move.py#L322

Reply:
This should be done in account itself, not in purchase.  You are super-confusing me.

Reply:
And the fiscal position should be copied when you create the vendor bill from the purchase.  We don't create a vendor bill on the fly and link it then with a purchase (to my knowledge)

Reply:
This PR is totally wrong, but there might be things to verify.

Reply:
But it's not copied from the PO, as I explained in the commit message.
The reason is, when creating the bill some fields are written on the account move, and then the onchange methods are called, because of that the fiscal position is overriden by this onchange https://github.com/odoo/odoo/blob/27edc28a2c19fad80053e668a8c72e7a629e4c1a/addons/account/models/account_move.py#L368-L371

Reply:
@guva-odoo But then, as @william-andre said, there is a problem that this onchange should not override if the fiscal position has already been set by the purchase order.  Which could be something in account onchanges being done wrong and we need to find where/why it is changed.

Reply:
I don't understand a thing about this conversation nor can I reproduce the difference explained in the ticket between community and invoicing behaviour. Can anyone take a little height and explain to me as if I were a 5 year old child ?

Reply:
@tsb-odoo when you create a vendor bill from a purchase order, the fiscal position should be the one from the vendor bill, but somehow it gets recalculated (which mostly gives the same, but the user could have changed it on the vendor bill) (see description at the beginning)

Reply:
... the one from the PO... but my concern is ... why then don't we change the FP of the partner ? Is it a valid and challenged use case that a PO would have a different FP than the partner's ?

Reply:
@tsb-odoo I would think so because you could buy something in your Belgian company and let it arrive in your warehouse in Spain.  Does not this affect the taxes?

Reply:
> ... the one from the PO... but my concern is ... why then don't we change the FP of the partner ?

We wouldn't have a fiscal position field on the invoices/PO/SO if it was always the fiscal position of the partner